packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts

```diff
@@ -299,6 +305,7 @@ export const updateEntityResolver: ResolverFn<
       { graphApi },
       {
         entity,
+        entityTypeId: entityTypeId ?? undefined,
```

Comment:
What is the type of `entityTypeId` (the variable, not the field in the object) here? If it's `VersionedUri | Undefined` do we need `?? undefined`?

Reply:
It's `InputMaybe<`${string}/v/${number}`> | undefined`, so there is a `null` difference between both. field is not expecting null, so I used `?? undefined` to make sure it's not getting `null`. Same approach used in several places, even in the same file above. See https://github.com/hashintel/hash/blob/3cd987b35571126b1b3c012f4119d0e347b33e82/packages/hash/api/src/graphql/resolvers/knowledge/entity/entity.ts#L293-L294

Reply:
Makes sense, cheers